apis/dscinitialization/v1/dscinitialization_types.go

```diff
@@ -35,9 +35,22 @@ type DSCInitializationSpec struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=2
 	// +optional
 	Monitoring Monitoring `json:"monitoring,omitempty"`
+	// Configures Service Mesh as networking layer for Data Science Clusters components.
+	// The Service Mesh is a mandatory prerequisite for single model serving (KServe) and
+	// you should review this configuration if you are planning to use KServe.
+	// For other components, it enhances user experience; e.g. it provides unified
+	// authentication giving a Single Sign On experience.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=3
+	// +optional
+	ServiceMesh ServiceMeshSpec `json:"serviceMesh,omitempty"`
+	// Configures Serverless (KNative Serving). This is a prerequisite for single model
+	// serving (KServe) and you should review this configuration if you are planning to use KServe.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=4
+	// +optional
+	Serverless ServerlessSpec `json:"serverless,omitempty"`
```

Comment:
It is... Do you mean this should be named `KServe` or something else?

If it should be named `KServe` probably it will become weird, as there will be the same field in the `DSC` which is the one that would actually control KServe deployment...

Whatever we do here, I think it will always be weird. We could change it, but then Service Mesh is also an implementation detail for KServe with unified auth being the only thing that is augmenting it as the base dependency.

Reply:
Any thoughts on this @VaishnaviHire @zdtsw?

Reply:
if i understand it correctly:
- serverless is only for kserve (for now), 
- servicemesh can be used for all components (sooner but start with kserve) and it will be done by toggle in each component
the concern here is, if we should keep serverless in DSCI or move to DSC on the kserve component level (and ofc keep servicemesh in DSCI) ?

Reply:
> the concern here is, if we should keep serverless in DSCI or move to DSC on the kserve component level (and ofc keep servicemesh in DSCI) ?

It could be a way to solve it, although I don't think timing is good for this, as I think moving to DSC would be a different approach, given the whole `Feature` structs are around tied to the DSCI as the owner (including `FeatureTrackers`).

> I do wonder ... in theory serverless is an implementation detail for `KServe` component... so it is a bit ambiguous to name it this way.

That ^ is what I think is the concern. Is more about if `Serverless` is a good naming.

Reply:
Yea DSCI crd was setup to include common settings applied to all the components. Serverless doesn't fulfill that requirement.

Wouldn't we just need to duplicate the `FeatureTrackers` struct in `./components/component.go` , similar to Devflags?

Reply:
My comment was about naming, but I think it's worth addressing the concern about `serverless` being cross-cutting component or exclusive for KServe.


> Yea DSCI crd was setup to include common settings applied to all the components. Serverless doesn't fulfill that requirement.

So the proposal here is to move this to KServe component?  If we do not foresee this as a common infra component and it is only ever needed for KServe we should definitely do that. Also - moving it there is not a big effort. Features and FeatureTrackers do not have to be moved around at all nor duplicated (you can see I use them in both dsci and components in #605 - eg. for Dashboard istio resources). It boils down to where we draw the line.

Reply:
I think what we will need to change is to make `featuretrackers.opendatahub.io` CRD a top-level one instead of making them part of DSCI.

Reply:
As discussed during our call last week - this has been reworked so that serverless config is part of KServe component now, not DSCI.